File: 2019/day1.py
def read_input(inp):
    f = open(inp, "r")
    l = f.readlines()
    for i in range(len(l)):
        l[i] = l[i].strip("\n")
    return l


l = read_input("day1.txt")

# part 1
s = 0
for i in l:
    i = int(i)
    s += (i / 3) - 2
print s


# part 2
f = 0
s = 0
for i in l:
    i = int(i)
    f = (i / 3) - 2
    while f > 0:
        s += f
        f = (f / 3) - 2

print s

File: 2021/day1.go
package main

import (
	"2021/util"
	"fmt"
	"strconv"
)

func Day1() {
	fmt.Printf("\nDay 1: \n")
	stringData := util.ReadInput("./day1.txt")

	// make slice of ints
	intData := make([]int64, len(stringData))
	for i, v := range stringData {
		intData[i], _ = strconv.ParseInt(v, 10, 64)
	}

	partOneAns := 0
	for i := 1; i < len(intData); i++ {
		if intData[i] > intData[i-1] {
			partOneAns++
		}
	}
	fmt.Printf("Part 1: %d\n", partOneAns)

	partTwoAns := 0
	for i := 3; i < len(intData); i++ {
		if intData[i] > intData[i-3] {
			partTwoAns++
		}
	}
	fmt.Printf("Part 2: %d\n", partTwoAns)
}

